e2e/client/ippool.go

```diff
@@ -16,8 +14,7 @@ import (
 func isIPPoolAllocationsEmpty(k8sIPAM *kubeClient.KubernetesIPAM, ipPoolName string) wait.ConditionFunc {
 	return func() (bool, error) {
 		ipPool, err := k8sIPAM.GetIPPool(context.Background(), ipPoolName)
-		noPoolError := fmt.Errorf("k8s pool initialized")
-		if errors.As(err, &noPoolError) {
+		if err.Error() == kubeClient.ErrKMsg8sPoolInit {
```

Comment:
what's the advantage of comparing strings vs using the golang errors API ? I mean, why don't we just fix the `errors.As` call ? 

According to golang's release notes, go vet now just "reports when [errors.As](https://tip.golang.org/pkg/errors/#As) is called with a second argument of type *error, a common mistake."

Reply:
> This makes the reviewers life harder, and renders tools like `git bisect` useless if this happens to be merged.

I agree, it took a while to run all the tests locally and fix.